### PR TITLE
task/02-interfaces

### PR DESCRIPTION
**Pull request by SerhiiTrokhymets**
**Acceptance criteria for 1**

- [x]  Mentor is added to pr
- [x]  AC copied to pr description
- [x]  Link to deployed version is added to pr
- [x]  Interfaces for Product and Category based on [swagger](https://ma-backend-api.mocintra.com/docs#/categories/CategoriesController_getAll) are created and exported in separate file.
- [x]  Static header and footer are created according to [mobile](https://www.figma.com/file/mRAOMdJiAsD8fHSVnYwUl9/React-web-page?type=design&node-id=13-17490&mode=design&t=3Thf45kJJYMoxsc0-0) and [desktop](https://www.figma.com/file/MIKTcyr6SvspkZnIQ5nEPD/MA-Courses-2022-2023?type=design&node-id=2471-2253&mode=design&t=4DJ0BIdFZZ4mUgAn-0) designs.

**Acceptance criteria for 1***

- [ ]  Shared component SocialIcon is created and used for footer icons
- [ ]  Link to social network and path to icon are passed to component by the typed props

### CODE REVIEW

File: src/components/footer/Footer.component.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { IconsComponent } from '@/components/icons/Icons.component.tsx';
+
+import footerStyles from './footer.module.css';
+
+export const FooterComponent = () => (
+    <div className={footerStyles.container}>
+        <div className={footerStyles.divider}>
+            <svg xmlns="http://www.w3.org/2000/svg" width="1148" height="2" viewBox="0 0 1148 2" fill="none">
```
Comment (serhii-tsivinskyi-moc) [inline src/components/footer/Footer.component.tsx]: Using CSS to create simple elements like dividers is often a better approach for better consistency and easier maintenance.

File: src/components/api/apiInterfaces.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+export interface Product {
+    id: number;
+    title: string;
+    price: number;
+    description: string;
+    images: Array<string>;
+    creationAt: Date;
```
Comment (serhii-tsivinskyi-moc) [inline src/components/api/apiInterfaces.ts]: it should be a string, not a Date.

File: src/components/api/apiInterfaces.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+export interface Category {
+    id: number;
+    name: string;
+    image: string;
+    creationAt: Date;
+    updatedAt: Date;
+}
+
+export interface Products extends Array<Product> {}
```
Comment (serhii-tsivinskyi-moc) [inline src/components/api/apiInterfaces.ts]: in this case more better to use types

File: src/components/footer/Footer.component.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+                <path d="M0 1H1148" stroke="#CCCCCC" />
+            </svg>
+        </div>
+        <>
+            <IconsComponent />
+        </>
+        <h2 className={footerStyles.author}>
+            Made with 💗 on course{' '}
+            <a href="https://www.mastersacademy.education/frontend-for-beginners-it" className={footerStyles.redColor}>
```
Comment (serhii-tsivinskyi-moc) [inline src/components/footer/Footer.component.tsx]: Assign CSS class names based on their role in the HTML structure

File: src/components/icons/Icons.component.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import facebook from '../../assets/facebook.svg';
+import instagram from '../../assets/instagram.svg';
+import linkedin from '../../assets/linkedin.svg';
+
+import styles from './icons.module.css';
+
+export const IconsComponent = () => (
```
Comment (serhii-tsivinskyi-moc) [inline src/components/icons/Icons.component.tsx:7]: 3. Implement a dynamic FooterIcon component that includes two props: an icon and a URL. This component should display the passed data as per the design. Use this component to display icons in the footer. Don't forget about typing.

File: src/components/footer/footer.module.css [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  font-family: Montserrat, serif;
+  font-size: 16px;
+  font-style: normal;
+  font-weight: 400;
+  text-align: center;
+  margin: 0;
+}
+
+.redColor {
+  color: #EF4934;
```
Comment (serhii-tsivinskyi-moc) [inline src/components/footer/footer.module.css]: use css variables for all colors

File: src/components/api/apiInterfaces.ts (hunk 3 above)
Comment (SerhiiTrokhymets) [reply src/components/api/apiInterfaces.ts]: done

File: src/components/api/apiInterfaces.ts (hunk 2 above)
Comment (SerhiiTrokhymets) [reply src/components/api/apiInterfaces.ts]: done

File: src/components/footer/footer.module.css (hunk 6 above)
Comment (SerhiiTrokhymets) [reply src/components/footer/footer.module.css]: done

File: src/components/footer/Footer.component.tsx (hunk 4 above)
Comment (SerhiiTrokhymets) [reply src/components/footer/Footer.component.tsx]: done

Comment (serhii-tsivinskyi-moc) [reply src/components/footer/Footer.component.tsx]: Here I mean to rename `redColor`

Comment (SerhiiTrokhymets) [reply src/components/footer/Footer.component.tsx]: done

File: src/components/footer/Footer.component.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+import footerStyles from './footer.module.css';
+
+export const FooterComponent = () => (
+    <div className={footerStyles.container}>
+        <div className={footerStyles.divider}>
+            <svg xmlns="http://www.w3.org/2000/svg" width="auto" height="2" viewBox="0 0 1148 2" fill="none">
+                <path d="M0 1H1148" stroke="#CCCCCC" />
+            </svg>
+        </div>
+        <>
```
Comment (serhii-tsivinskyi-moc) [inline src/components/footer/Footer.component.tsx]: you don't need this tag here

File: src/components/footer/Footer.component.tsx [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import { IconsComponent } from '@/components/icons/Icons.component.tsx';
+
+import footerStyles from './footer.module.css';
+
+export const FooterComponent = () => (
+    <div className={footerStyles.container}>
+        <div className={footerStyles.divider}>
```
Comment (serhii-tsivinskyi-moc) [inline src/components/footer/Footer.component.tsx]: width of divider should match width of text content + image

File: src/App.tsx [hunk 9]
```diff
@@ -1,11 +1,15 @@
...
+import { FooterComponent } from '@/components/footer/Footer.component.tsx';
+import { HeaderComponent } from '@/components/header/Header.component.tsx';
 
 import './App.css';
 
 function App() {
     return (
         <>
+            <HeaderComponent />
             <AboutMeComponent />
```
Comment (serhii-tsivinskyi-moc) [inline src/App.tsx:11]: For wide screens (2000px+), center the content of the "about" section on the page.

File: src/components/footer/Footer.component.tsx (hunk 4 above)
Comment (serhii-tsivinskyi-moc) [reply src/components/footer/Footer.component.tsx]: Name the class "link". Even if your class has only one style, it's best to follow standard naming conventions.

Comment (SerhiiTrokhymets) [reply src/components/footer/Footer.component.tsx]: @serhii-tsivinskyi-moc rename to 'link'

File: src/components/footer/Footer.component.tsx (hunk 7 above)
Comment (SerhiiTrokhymets) [reply src/components/footer/Footer.component.tsx]: @serhii-tsivinskyi-moc removed unnecessary tags

File: src/App.tsx (hunk 9 above)
Comment (SerhiiTrokhymets) [reply src/App.tsx:11]: @serhii-tsivinskyi-moc done

File: src/components/footer/Footer.component.tsx (hunk 1 above)
Comment (SerhiiTrokhymets) [reply src/components/footer/Footer.component.tsx]: done

File: src/components/footer/Footer.component.tsx (hunk 8 above)
Comment (serhii-tsivinskyi-moc) [reply src/components/footer/Footer.component.tsx]: width of divider still doesn't match a design

**Review by serhii-tsivinskyi-moc — APPROVED** [review]

**Pull request merged**